src/core/lib/gprpp/per_cpu.h

```diff
@@ -29,16 +31,20 @@ namespace grpc_core {
 template <typename T>
 class PerCpu {
  public:
-  T& this_cpu() { return data_[ExecCtx::Get()->starting_cpu()]; }
+  explicit PerCpu(size_t max = std::numeric_limits<size_t>::max())
```

Comment:
> What's the purpose of this change? If there are actually more CPUs, don't we actually want that many more instances?
> If we do impose a limit on the total number of instances, what happens if this_cpu() gets called from a CPU that is greater than the limit? It seems like that would just cause a crash.

Thanks for catching this, fixed. Craig was concerned when there are too many cores. We actually did the same in Fathom in prod where we capped it at 16~32, so I picked 32 here, which I think will be fine.